### PR TITLE
Add support for icon-color

### PR DESCRIPTION
**Pull request by bartvde**
This adds support for ```icon-color``` on sprite images

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -480,16 +483,51 @@ export default function(olLayer, glStyle, source, resolutions, spriteData, sprit
...
                 if (!iconImg) {
                   var spriteImageData = spriteData[icon];
-                  iconImg = iconImageCache[icon] = new Icon({
-                    img: spriteImage,
-                    imgSize: spriteImgSize,
-                    size: [spriteImageData.width, spriteImageData.height],
-                    offset: [spriteImageData.x, spriteImageData.y],
-                    scale: paint['icon-size'](zoom, properties) / spriteImageData.pixelRatio
-                  });
+                  if (iconColor[0] !== 0 || iconColor[1] !== 0 || iconColor[2] !== 0) {
```
Comment (ahocevar) [inline index.js]: This would be more efficient if there was no default for `icon-color`.

File: index.js [hunk 2]
```diff
@@ -480,16 +483,51 @@ export default function(olLayer, glStyle, source, resolutions, spriteData, sprit
                   style = styles[stylesLength] = new Style();
                 }
                 style.setGeometry(styleGeom);
-                iconImg = iconImageCache[icon];
+                var iconSize = paint['icon-size'](zoom, properties);
+                var iconColor = paint['icon-color'](zoom, properties);
+                var icon_cache_key = icon + '.' + iconSize + '.' + iconColor;
+                iconImg = iconImageCache[icon_cache_key];
```
Comment (ahocevar) [inline index.js]: Is `icon-size` supported or used at all for non-circle icons?

Comment (bartvde) [reply index.js]: yeah it is see: https://www.mapbox.com/mapbox-gl-js/style-spec/#layout-symbol-icon-size

it works, although ofcourse it is not very sharp

File: index.js (hunk 1 above)
Comment (bartvde) [reply index.js]: agreed, I took the default from the mapbox style spec, but indeed it makes more sense to have no default, will change

Comment (ahocevar) [reply index.js]: Right. A default would make sense for sdf icons, but we don't have those, and png icons have their own color already.

Comment (bartvde) [reply index.js]: See: https://github.com/boundlessgeo/mapbox-to-ol-style/pull/20/commits/cde172bd69750a661ccf84da541a77f150b48f1e

**Review by ahocevar — APPROVED** [review]
Looks great now. Please merge.

**Comment by bartvde** [comment]
thanks for the review @ahocevar

**Pull request merged**